sox/transform.py

```diff
@@ -1820,6 +1828,73 @@ def mcompand(self, n_bands=2, crossover_frequencies=[1600],
         self.effects_log.append('mcompand')
         return self
 
+    def noiseprof(self, profile_path):
+        '''Calculate  a  profile  of  the audio for use in noise reduction.
+        See the description of the noisered effect for details.
+
+        Paramters
+        ---------
+        profile : str, default='noise.prof'
+            A file used to store noise inforation.
+
+        profile_path : str
+            Path to save the noise profile file.
+
+        See Also
+        --------
+        noisered
+        '''
+
+        try:
+            open(profile_path, 'w')
+        except IOError:
+            raise IOError("profile_path {} is not writeable.".format(profile_path))
+        else:
```

Comment:
Well, I feels that both `profile_path` and  `profile` are needed. I added them in new version.